packages/mobile/src/harmony-native/components/Radio/useRadioGroup.tsx

```diff
@@ -0,0 +1,27 @@
+import { useCallback, useContext } from 'react'
+
+import { RadioGroupContext } from './RadioGroupContext'
+
+type RadioGroupProps = {
+  checked?: boolean
+  value?: any
+  onValueChange?: (value: string) => void
+}
+
+export const useRadioGroup = (props: RadioGroupProps) => {
+  const { checked: checkedProp, value, onValueChange } = props
+  const radioGroup = useContext(RadioGroupContext)
+  let checked = checkedProp
+  if (radioGroup) {
```

Comment:
I assume this can only be undefined because there might be no context?

Would it be overly aggro to just throw an error if you call `useRadioGroup` outside a radio context?
Probably worth a warning at least imo

Reply:
though I see perhaps that this isn't really meant for consumption outside the Radio implementation

Reply:
yes great points.... id like us to use the same implementation on web too, so logging a ticket, id like to see what marcus thinks too since he was original implementor